server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -255,8 +255,12 @@ public XContentBuilder innerToXContent(XContentBuilder builder, Params params) t
     }
 
     public static SearchResponse fromXContent(XContentParser parser) throws IOException {
-        ensureExpectedToken(Token.START_OBJECT, parser.nextToken(), parser::getTokenLocation);
-        parser.nextToken();
```

Comment:
Can we try to avoid this change ? Maybe using a custom parser in the root object parser for the `response` field ?

Reply:
I did that at first but found the change in the server object is cleaner. Happy to change to the other solution though, but I think its working around something that we can fix better here.

Reply:
We can also split into `fromXContent` and `fromXContentFragment`, the former would consume the start object and delegate to `fromXContentFragment` ?

Reply:
We already have `innerFromXContent` which in an earlier version of this PR I exposed and used from the client side, but then we have to explain that there. I'll change the PR back to that version so you can see the options.